File: 2154-keep-multiplying-found-values-by-two/2154-keep-multiplying-found-values-by-two.cs
// this can be made faster if we hash all numbers and then check if num*2 is present in hash
public class Solution {
    public int FindFinalValue(int[] nums, int original) {
        Array.Sort(nums);
        foreach(var num in nums){
            if(num == original) original*=2;
        }
        return original;
    }
}

File: 2154-keep-multiplying-found-values-by-two/2154-keep-multiplying-found-values-by-two.java
// this can be made faster if we hash all numbers and then check if num*2 is present in hash
class Solution {
    public int findFinalValue(int[] nums, int original) {
        Arrays.sort(nums);
        for(var num : nums)
            if(num == original) original*=2;
        return original;
    }
}